document.go

```diff
@@ -83,6 +83,14 @@ type GeoShapeField interface {
 	GeoShape() (GeoJSON, error)
 }
 
+type DenseVectorField interface {
```

Comment:
Separate file with build tag.

Reply:
Let's also rename this interface to `VectorField`

Reply:
Also  - what is the need for this interface, just introducing the struct in bleve should suffice yes?

Reply:
Zap usage, nevermind last question.